src/DataStructures/Tensor/Expressions/ScaleBy.hpp

```diff
@@ -0,0 +1,31 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines function scale_by
+
+#pragma once
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+
+/*!
+ * \ingroup Functions
+ * \brief scales a tensor by a scalar
+ *
+ * \returns a tensor of the same type as the input tensor
+ *
+ * \details
+ * Divides the input tensor by the given scalar.
+ */
+template <typename TensorType>
+TensorType scale_by(const TensorType& tensor, const DataVector& scale) {
+  ASSERT(tensor.get(0).size() == scale.size(),
+         "The size of the DataVector in 'tensor' is not the same as the size of "
```

Comment:
"size of scale" makes me think of the magnitude, not the number of elements in `scale` - this can be fixed with either a rewording of the assert message (which currently only restates the == check that failed) or again, by changing the name `scale`